dbdimp.c

```diff
@@ -3855,10 +3854,14 @@ my_ulonglong mariadb_st_internal_execute41(
       mariadb_dr_do_error(sth, mysql_errno(svsock), mysql_error(svsock), mysql_sqlstate(svsock));
       return -1;
     }
+    if (mysql_stmt_prepare(stmt, sbuf, slen))
+    {
+      mariadb_dr_do_error(sth, mysql_stmt_errno(stmt), mysql_stmt_error(stmt), mysql_stmt_sqlstate(stmt));
+      mysql_stmt_close(stmt);
```

Comment:
This will not call ``mysql_stmt_close(*stmp_ptr)`` (previously it was called) - is it intentional?

Reply:
Yes, `*stmt_ptr` holds `MYSQL_STMT` structure which was successfully prepared and therefore is in internal prepared state (and it should not cause that CVE 2017-3302 crash). Structure is valid and any `mysql_stmt_*` function can be called on it.

After this change, any call to DBD::MariaDB should let `imp_sth->stmt` pointer either to NULL or successfully prepared `MYSQL_STMT` structure.